File: Least Number of Unique Integers after K Removals.go
func findLeastNumOfUniqueInts(arr []int, k int) int {
    counter := make(map[int]int)
    for _, num := range arr {
        if _, ok := counter[num]; ok {
            counter[num] += 1
        } else {
            counter[num] = 1
        }
    }

    n := len(counter)
    counts := make([]int, n)
    j := 0
    for _, count := range counter {
        counts[j] = count
        j++
    }
    slices.Sort(counts)

    for i, count := range counts {
        if k < count {
            return n-i
        } else {
            k -= count
        }
    }
    return 0
}

File: Least Number of Unique Integers after K Removals.cpp
class Solution {
public:
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        unordered_map<uint32_t, uint16_t> counter;
        for (auto & num: arr) {
            if (counter.find(num) == counter.end()) {
                counter.insert(make_pair(num, 1));
            } else {
                counter.at(num)++;
            }
        }

        vector<uint16_t> counts;
        int n = counter.size();
        counts.reserve(n);
        for (auto & iter: counter) {
            counts.push_back(iter.second);
        }
        sort(counts.begin(), counts.end());
        for (int i = 0; i < n; i++) {
            if (k < counts[i]) {
                return n-i;
            } else {
                k -= counts[i];
            }
        }
        return 0;
    }
};
